common/src/main/java/com/bakdata/dedupe/clustering/RefineCluster.java

```diff
@@ -139,7 +139,7 @@ private static double scoreClustering(final byte[] partitions, final double[][]
                 .mapToObj(i -> {
                     // reverse of Gaussian
                     int leftIndex = (int) (Math.sqrt(i + 0.25) - 0.5);
-                    int rightIndex = i - getNumEdges(leftIndex) + leftIndex;
```

Comment:
Second error, it should just be int rightIndex = i - getNumEdges(leftIndex);

Reply:
Explanation: leftIndex and rightIndex should take all possible values where leftIndex > rightIndex and leftIndex < numNodes and rightIndex < numNodes.
i is a random edge < numEdges
leftIndex is the largest node, such that leftIndex * (leftIndex + 1) / 2 <= i
rightIndex is then the remainder.

Examples:

i | leftIndex | rightIndex
--- | --- | ---
0 | 0 | 0
1 | 1 | 0
2 | 1 | 1
3 | 2 | 0
4 | 2 | 1
5 | 2 | 2
6 | 3 | 0
7 | 3 | 1
8 | 3 | 2
9 | 3 | 3
10 | 4 | 0
11 | 4 | 1
12 | 4 | 2
13 | 4 | 3
14 | 4 | 4
15 | 5 | 0
16 | 5 | 1
17 | 5 | 2
18 | 5 | 3
19 | 5 | 4
20 | 5 | 5
21 | 6 | 0
22 | 6 | 1
23 | 6 | 2
24 | 6 | 3
25 | 6 | 4
26 | 6 | 5
27 | 6 | 6